ts/features/bonus/cgn/saga/networking/details/getCgnInformationSaga.ts

```diff
@@ -0,0 +1,29 @@
+import { call, put } from "redux-saga/effects";
+import { readableReport } from "italia-ts-commons/lib/reporters";
+import { BackendCGN } from "../../../api/backendCgn";
+import { SagaCallReturnType } from "../../../../../../types/utils";
+import { CgnActivatedStatus } from "../../../../../../../definitions/cgn/CgnActivatedStatus";
+import { cgnDetails } from "../../../store/actions/details";
+
+export function* cgnGetInformationSaga(
+  getCgnStatus: ReturnType<typeof BackendCGN>["getCgnStatus"]
+) {
+  try {
+    const cgnInformationResult: SagaCallReturnType<typeof getCgnStatus> = yield call(
+      getCgnStatus,
+      {}
+    );
+    if (cgnInformationResult.isLeft()) {
+      throw Error(readableReport(cgnInformationResult.value));
+    } else if (
+      cgnInformationResult.isRight() &&
+      CgnActivatedStatus.is(cgnInformationResult.value.value)
+    ) {
+      yield put(cgnDetails.success(cgnInformationResult.value.value));
+    } else {
+      throw Error("Card is in another status");
```

Comment:
This `Error` will throw even if the response have a status code !== 200.
So the error message could be misleading

Considering instead to have an accurate error handling to have a strong vision of what is happing

```typescript
export function* cgnGetInformationSaga(
  getCgnStatus: ReturnType<typeof BackendCGN>["getCgnStatus"]
) {
  try {
    const cgnInformationResult: SagaCallReturnType<typeof getCgnStatus> = yield call(
      getCgnStatus,
      {}
    );
    if (cgnInformationResult.isLeft()) {
      throw Error(readableReport(cgnInformationResult.value));
    } else if (
      cgnInformationResult.isRight() &&
      cgnInformationResult.value.status === 200
    ) {
      if (CgnActivatedStatus.is(cgnInformationResult.value.value)) {
        yield put(cgnDetails.success(cgnInformationResult.value.value));
      } else {
        throw Error(
          `Card is in another status ${cgnInformationResult.value.value.status}`
        );
      }
    } else {
      throw Error(`response status ${cgnInformationResult.value.status}`);
    }
  } catch (e) {
    yield put(cgnDetails.failure(getError(e)));
  }
}
```

Reply:
Since we should show the CGN detail screen even if it is in Revoked or Expired status we can remove the check and throw an error in any case except the status 200 what do you think?

here 5c77855 the solutions